core_manager/modules/modem.py

```diff
@@ -213,12 +213,25 @@ def initiate_ecm(self):
     def check_interface_health(self, interface, timeout):
         output = shell_command(f"ping -q -c 1 -s 8 -w {timeout} -I {interface} 8.8.8.8")
 
-        if output[2] == 0:
-            pass
-        else:
+        if output[2] != 0:
             raise NoInternet("No internet!")
 
+    def update_cellular_interface_name(self):
+        output = shell_command("lshw -C Network")
+
+        if output[2] == 0:
+            networks = output[0].split("*-network")
+
+            for network in networks:
+                if network.find("driver=cdc_ether") >= 0:
+                    for line in network.splitlines():
+                        line = line.lstrip()
+                        vals = line.split(': ')
+                        if len(vals) >= 2 and vals[0] == "logical name":
+                            self.interface_name = vals[1]
+
     def check_internet(self):
+        self.update_cellular_interface_name()
```

Comment:
The network manager module must decide the interface name due to our architecture. So we don't prefer to give `update_cellular_interface_name` task to the connection manager module. 

NM checks if any change in interface name every 30 seconds. Internet checking mechanism runs every 60 seconds. So, this is not a big deal to make us increase the complexity of the code.